x/accounts/keeper.go

```diff
@@ -107,6 +107,23 @@ func (k Keeper) IsAccountsModuleAccount(
 	return hasAcc
 }
 
+func (k Keeper) NextAccountNumber(
+	ctx context.Context,
+) (accNum uint64, err error) {
+	accNum, err = k.AccountNumber.Next(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	return accNum, nil
+}
+
+// only use to migrate account number
+func (k Keeper) SetAccountNumber(ctx context.Context, accNum uint64) error {
+	err := k.AccountNumber.Set(ctx, accNum)
```

Comment:
You can also have the sanity check here that ensures that the new value is => old value.  The method is public so better safe than sorry